tokio/src/runtime/blocking/pool.rs

```diff
@@ -187,13 +195,17 @@ impl Spawner {
         };
 
         if let Some(shutdown_tx) = shutdown_tx {
-            self.spawn_thread(shutdown_tx, rt);
+            let handle = self.spawn_thread(shutdown_tx, rt);
+
+            let mut shared = self.inner.shared.lock().unwrap();
+            shared.worker_threads.push(handle);
+            drop(shared);
```

Comment:
Nit, take it or leave it: I'm not convinced the explicit drop of `shared` is really necessary or makes the code all that much clearer. This could just be:
```suggestion
            self.inner.shared.lock().unwrap().worker_threads.push(handle);
```

Reply:
I personally like the "critical section" feeling of it, but for a one-liner, I changed it. It now has a "atomic" feeling to it.